StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -25,17 +27,49 @@ public class SA1008OpeningParenthesisMustBeSpacedCorrectly : DiagnosticAnalyzer
         /// The ID for diagnostics produced by the <see cref="SA1008OpeningParenthesisMustBeSpacedCorrectly"/> analyzer.
         /// </summary>
         public const string DiagnosticId = "SA1008";
+
+        internal const string LocationKey = "location";
+        internal const string ActionKey = "action";
+        internal const string LocationPreceding = "preceding";
+        internal const string LocationFollowing = "following";
+        internal const string ActionInsert = "insert";
+        internal const string ActionRemove = "remove";
+
         private const string Title = "Opening parenthesis must be spaced correctly";
-        private const string MessageFormat = "Opening parenthesis must{0} be {1} by a space.";
         private const string Category = "StyleCop.CSharp.SpacingRules";
         private const string Description = "An opening parenthesis within a C# statement is not spaced correctly.";
         private const string HelpLink = "http://www.stylecop.com/docs/SA1008.html";
 
-        private static readonly DiagnosticDescriptor Descriptor =
-            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, AnalyzerConstants.DisabledNoTests, Description, HelpLink);
+        private const string MessageNotPreceded = "Opening parenthesis must not be preceded by a space.";
+        private const string MessagePreceded = "Opening parenthesis must be preceded by a space.";
+        private const string MessageNotFollowed = "Opening parenthesis must not be followed by a space.";
 
         private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
-            ImmutableArray.Create(Descriptor);
+            ImmutableArray.Create<DiagnosticDescriptor>()
```

Comment:
:bulb: This is inefficient, and also unnecessary. You only need to return one descriptor per diagnostic ID. Also, the following factory method would be preferred because it only creates one array:

```cs
ImmutableArray.Create(descriptor1, descriptor2, descriptor3)
```